pkg/controller/jobframework/reconciler.go

```diff
@@ -368,7 +369,29 @@ func (r *JobReconciler) equivalentToWorkload(job GenericJob, object client.Objec
 	if owner.Name != object.GetName() {
 		return false
 	}
-	return job.EquivalentToWorkload(*wl)
+
+	jobPodSets := job.PodSets()
+
+	if !workload.CanBePartiallyAdmitted(wl) || job.IsSuspended() {
```

Comment:
mental note for self:

Maybe we can change `IsSuspended()` for `batch/Job` to `suspend == true && startTime == nil`